app/src/lib/dispatcher/dispatcher.ts

```diff
@@ -841,6 +841,20 @@ export class Dispatcher {
     return this.appStore._setConfirmRepoRemoval(value)
   }
 
+  public openInExternalEditor(repository: Repository, filepath: string): boolean {
+    const fullPath = Path.join(repository.path, filepath)
+    // because Windows uses different path separators here
+    const normalized = Path.normalize(fullPath)
```

Comment:
Regarding this question:

> should I refactor the path normalization code into a helper function? I just copied and pasted it from the repo clone code from the same file.

While it's rather trivial, I think extracting this to a local function would be :gem:.

I went with this, but you can totally try a different result if you find something better:

```ts
function normalizePath(repository: Repository, filepath: string): string {
  const fullPath = Path.join(repository.path, filepath)
  // because Windows uses different path separators here
  const normalized = Path.normalize(fullPath)
  return normalized
}
```

This then makes the usage:

```ts
const path = normalizePath(repository, filepath)
shell.openItem(path)
```